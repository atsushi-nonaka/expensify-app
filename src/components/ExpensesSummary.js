import React from 'react'
import { connect } from 'react-redux'
import expenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'


export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expensesCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
        </div>
    )
}    
const mapStateToProps = (state) => {
    const visibleExpenses = expenses(state.expenses, state.filters) 
    return {
        expensesCount: visibleExpenses.length, 
        expensesTotal: expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)