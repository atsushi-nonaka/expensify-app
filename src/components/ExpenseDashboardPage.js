import React from 'react'
import ExpenseList from './ExpenseList'
import ExpsenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpsenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage