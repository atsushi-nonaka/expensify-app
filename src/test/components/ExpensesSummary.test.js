import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should correctly render ExpenseSummary with a single expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={123000} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpenseSummary with a multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={453000} />)
    expect(wrapper).toMatchSnapshot()
})
