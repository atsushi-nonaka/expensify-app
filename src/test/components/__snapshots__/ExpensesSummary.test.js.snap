// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with a multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $4,530.00
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with a single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1,230.00
  </h1>
</div>
`;
